compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3003,76 +3011,315 @@ private BLangExpression createConditionFromMatchPatterns(List<BLangMatchPattern>
+        }
+        // If some patterns are not implemented, those should be detected before this phase
+        // TODO : Remove this after all patterns are implemented
+        return null;
+    }
+
+    private BLangExpression createConditionForErrorMatchPattern(BLangErrorMatchPattern errorMatchPattern,
+                                                                BLangSimpleVarRef matchExprVarRef) {
+
+        BType matchPatternType = errorMatchPattern.type;
+        DiagnosticPos pos = errorMatchPattern.pos;
+
+        BLangSimpleVariableDef resultVarDef = createVarDef("errorPatternResult$",
+                symTable.booleanType, null, pos);
+        BLangSimpleVarRef resultVarRef = ASTBuilderUtil.createVariableRef(pos, resultVarDef.var.symbol);
+        BLangBlockStmt mainBlockStmt = ASTBuilderUtil.createBlockStmt(pos);
+        mainBlockStmt.addStatement(resultVarDef);
+
+        BLangAssignment failureResult =
+                ASTBuilderUtil.createAssignmentStmt(pos, resultVarRef, getBooleanLiteral(false));
+        BLangAssignment successResult =
+                ASTBuilderUtil.createAssignmentStmt(pos, resultVarRef, getBooleanLiteral(true));
+        mainBlockStmt.addStatement(failureResult);
+
+        BLangExpression typeCheckCondition = createIsLikeExpression(errorMatchPattern.pos, matchExprVarRef,
+                matchPatternType);
+        BLangExpression typeConvertedExpr = addConversionExprIfRequired(matchExprVarRef, matchPatternType);
+        BLangSimpleVariableDef tempCastVarDef = createVarDef("$castTemp$", matchPatternType, typeConvertedExpr, pos);
+        BLangSimpleVarRef tempCastVarRef = ASTBuilderUtil.createVariableRef(pos, tempCastVarDef.var.symbol);
+
+        BLangBlockStmt ifBlock = ASTBuilderUtil.createBlockStmt(pos);
+        ifBlock.addStatement(tempCastVarDef);
+        BLangIf ifStmt = ASTBuilderUtil.createIfElseStmt(pos, typeCheckCondition, ifBlock, null);
+        mainBlockStmt.addStatement(ifStmt);
+
+        BLangExpression condition = null;
+        if (errorMatchPattern.errorMessageMatchPattern != null) {
+            DiagnosticPos messagePos = errorMatchPattern.errorMessageMatchPattern.pos;
+            BLangInvocation messageInvocation = createLangLibInvocationNode(ERROR_MESSAGE_FUNCTION_NAME,
+                    tempCastVarRef,
+                    new ArrayList<>(), null, messagePos);
+            BLangSimpleVariableDef messageVarDef = createVarDef("$errorMessage$", messageInvocation.type,
+                    messageInvocation, messagePos);
+            ifBlock.addStatement(messageVarDef);
+            BLangSimpleVarRef messageVarRef = ASTBuilderUtil.createVariableRef(messagePos,
+                    messageVarDef.var.symbol);
+            condition =
+                    createConditionForErrorMessageMatchPattern(errorMatchPattern.errorMessageMatchPattern,
+                            messageVarRef);
+        }
```

Comment:
Shall we reformat this code? Unnecessary line breaks it harder to read.

Reply:
Fixed